library/platform_util.c

```diff
@@ -103,7 +105,7 @@ void mbedtls_platform_zeroize( void *buf, size_t len )
 struct tm *mbedtls_platform_gmtime_r( const mbedtls_time_t *tt,
                                       struct tm *tm_buf )
 {
-#if defined(_WIN32) && !defined(EFIX64) && !defined(EFI32)
+#if defined(_WIN32) && !defined(PLATFORM_UTIL_USE_GMTIME)
     return( ( gmtime_s( tm_buf, tt ) == 0 ) ? tm_buf : NULL );
 #elif !defined(PLATFORM_UTIL_USE_GMTIME)
     return( gmtime_r( tt, tm_buf ) );
```

Comment:
There is a bug in this if/elif statement (not created by you, this is already there). Firstly `#if defined(_WIN32) && !defined(EFIX64) && !defined(EFI32)` is exactly the same as `!defined(PLATFORM_UTIL_USE_GMTIME)` so the if statement can be shortened to just `!defined(PLATFORM_UTIL_USE_GMTIME)`.

Secondly this means that the if condition is identical to the elif section and it will not and has never run. In terms of what it should be I'm not entirely sure because I'm not too familiar with what mingw32 does and does not support, from what I can tell neither `gmtime_s` or `gmtime_r`. Any other opinions would be appreciated on this.